.circleci/config.yml

```diff
@@ -26,13 +26,48 @@ jobs:
           name: code-coverage
           command: bash <(curl -s https://codecov.io/bash)
 
+      - run:
+          name: commitlint
+          command: |
+            echo "module.exports = {extends: ['@commitlint/config-conventional']}" > commitlint.config.js
+            npm install @commitlint/cli @commitlint/config-conventional
```

Comment:
if we decide to use this tool, it's probably better to add these as devDependencies in package.json instead of installing them here every time

Reply:
I want the config here to be agnostic of NPM projects hence why it is wired this way